### PR TITLE
Add methods to round polygon corners

### PR DESCRIPTION
**Pull request by PhilDeOrsey**
We added methods for computing bezier points, drawing a round polygon path using those points, and finally updated the drawSVG function to utilize them.

### CODE REVIEW

File: src/draw-svg.ts [hunk 1]
```diff
@@ -17,6 +18,7 @@ import {GeoElement} from './utilities';
 
 export type LineMark = 'bar'|'bar2'|'arrow'|'arrow2';
 export type LineArrow = 'start'|'end'|'both';
+const CIRCLE_MAGIC = 4*(Math.sqrt(2)-1)/3;
```
Comment (plegner) [inline src/draw-svg.ts:21]: Can you add a comment with a link to the source for this?

File: src/draw-svg.ts [hunk 2]
```diff
@@ -106,6 +108,67 @@ function drawArcArrows(x: Arc, type: LineArrow) {
...
+
+  const d1 = rad/length0;
+  const d2 = rad/length1;
+  const shift = 1 - CIRCLE_MAGIC;
+
+  const p1 = Point.interpolate(points[0], points[1], clamp(1 - d1, 0, 1));
+  const p2 = Point.interpolate(points[0], points[1], clamp(1 - d1*shift, 0, 1));
+  const p3 = Point.interpolate(points[1], points[2], clamp(d2*shift, 0, 1));
+  const p4 = Point.interpolate(points[1], points[2], clamp(d2, 0, 1));
+  const offsets = [p1, p2, p3, p4];
```
Comment (plegner) [inline src/draw-svg.ts:128]: Nit: You can just return the array directly, rather than assigning it to a variable

File: src/draw-svg.ts [hunk 3]
```diff
@@ -106,6 +108,67 @@ function drawArcArrows(x: Arc, type: LineArrow) {
...
+    first = `${points[0].x} ${points[0].y}`;
+  } else {
+    const p1 = points[points.length - 1];
+    const p2 = points[0];
+    const p3 = points[1];
+    const offsets = getBezierPoints([p1, p2, p3], radius);
+    first = `${offsets[3].x} ${offsets[3].y}`;
+  }
+
+  path += first; // move to the first point
```
Comment (plegner) [inline src/draw-svg.ts:148]: Nit: I don't think you need the "first" variable – you can just do path += `...` in the if above

**Review by plegner — APPROVED** [review]

File: src/draw-svg.ts (hunk 1 above)
Comment (PhilDeOrsey) [reply src/draw-svg.ts:24]: done

**Pull request merged**